File: Proyecto/Calendario_App/src/main/java/org/example/calendario_app/model/Etiqueta.java
package org.example.calendario_app.model;

public class Etiqueta {
    private int id_etiqueta;
    private String nombre;
    private String color;
    private Integer id_grupo;

    public Etiqueta() {
    }

    public Etiqueta(int id_etiqueta, String nombre, String color) {
        this.id_etiqueta = id_etiqueta;
        this.nombre = nombre;
        this.color = color;
    }

    public Etiqueta(int id_etiqueta, String nombre, String color, Integer id_grupo) {
        this.id_etiqueta = id_etiqueta;
        this.nombre = nombre;
        this.color = color;
        this.id_grupo = id_grupo;
    }

    public Etiqueta(String nombre, String color) {
        this.nombre = nombre;
        this.color = color;
    }

    public int getId() {
        return id_etiqueta;
    }

    public void setId(int id_etiqueta) {
        this.id_etiqueta = id_etiqueta;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public Integer getId_grupo() {
        return id_grupo;
    }

    public void setId_grupo(Integer id_grupo) {
        this.id_grupo = id_grupo;
    }
}

File: Proyecto_Android/Calendario_Android_App-31-01-2026/Calendario_Android_App-31-01-2026/Calendario_Android_App/app/src/main/java/com/example/calendario_android_app/model/Etiqueta.kt
package com.example.calendario_android_app.model

/**
 * Modelo de datos que representa una etiqueta o categoría para eventos.
 * Las etiquetas pueden ser personales (sin grupo) o compartidas (asociadas a un grupo).
 * 
 * @property idEtiqueta Identificador único de la etiqueta
 * @property nombre Nombre descriptivo de la etiqueta
 * @property color Color en formato hexadecimal (ej: "#FF5733") para identificación visual
 * @property idGrupo ID del grupo al que pertenece la etiqueta (null si es personal)
 */
data class Etiqueta(
    val idEtiqueta: Int = 0,
    val nombre: String,
    val color: String,
    val idGrupo: Int? = null
)
